File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String[] strNum = new String[numbers.length];
        
        for (int i = 0; i < numbers.length; i++) {
            strNum[i] = String.valueOf(numbers[i]);    
        }
        
        Arrays.sort(strNum, (o1, o2) -> (o2+o1).compareTo(o1+o2));
        
        // Arrays.sort(numbers, new Comparator<Integer>() {
        //     @Override
        //     public int compare(Integer o1, Integer o2) {
        //         return (Integer.parseInt(o2 + "" + o1)).compareTo(Integer.parseInt(o1 + "" + o2));
        //     }
        // });
        
        if (strNum[0].equals("0")) {
            return "0";
        }
        
        StringBuilder sb = new StringBuilder();
        for (String num : strNum) {
            sb.append(num);
        }
        return sb.toString();
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<int> numbers) {
    
    vector<string> str_numbers;
    for (int &number : numbers) {
        str_numbers.push_back(to_string(number));
    }
    
    sort(str_numbers.begin(), str_numbers.end(), [](const string &a, const string &b) {
        return a+b > b+a;
    });
    
    if (str_numbers[0] == "0") {
        return "0";
    }
    
    string answer;
    for (string &num : str_numbers) {
        answer += num;
    }
    
    return answer;
}